### PR TITLE
bump major to chef-13

### PR DESCRIPTION
**Pull request by lamont-granquist**
I had to manually hack things up a bit.  The released gems poise, poise-boiler and halite pin chef to `~> 12.0` so they won't solve and are commented out for now.  The released version of ohai does the same, essentially, via a pin on chef-config.  Ohai master has been relaxed, so we can point that at github.

### CODE REVIEW

**Comment by lamont-granquist** [comment]
i need to go get some food, i'll shave this yak further when i get back...

File: Gemfile [hunk 1]
```diff
@@ -11,6 +11,9 @@ source "https://rubygems.org"
 # of bundler versions prior to 1.12.0 (https://github.com/bundler/bundler/commit/193a14fe5e0d56294c7b370a0e59f93b2c216eed)
 gem "chef", path: "."
 
+# tracking master of ohai for chef-13.0 development, this should be able to be deleted after relase
```
Comment (shortdudey123) [inline Gemfile]: FYI... typo `relase`

**Comment by coderanger** [comment]
So my vote on the poise/halite stuff is that I change them to depend on `>= 12, < 14` and pull them in from git master instead of releases like we're doing with Ohai, at least during the transition.

**Comment by lamont-granquist** [comment]
That sounds fine, I suspect your vote is the only one that really matters there.

**Comment by coderanger** [comment]
Okay, that should be done now.

**Comment by coderanger** [comment]
@lamont-granquist What is the plan for this branch? Should we start doing 13-era changes here or are we going to merge it and then do those? I see you started compat-breaking fixes but I had thought this was just for the bump.

**Comment by lamont-granquist** [comment]
@coderanger we included triggers in the spec tests for chef 13 stuff that triggers as soon as i bump the version to 13.0.0, so to get to green, i have to actually do the chef-13 work.

i guess i could edit them all the pending or something and just smash merge then clean them up in some other PR, dunno, turns out i didn't think this day through very well...

**Comment by lamont-granquist** [comment]
REVIEW REQUEST:  Please keep the nitpicks and bikesheds low on this one.  I need to get this through in order to unblock master for Chef-13 work.  For example, I'm not real happy with the RELEASE_NOTES in their current form, but they're serving well enough as notes about what has been broken, and we have 2 months to fix them up to make them presentable to customers.  Once this gets merged then we're not serializing on this one PR and Chef-13 development all gets happier and more asynchronous.

**Comment by lamont-granquist** [comment]
@chef/client-core IDK what to do about travis but it was green other than unit tests recently and unit tests are all passing on my desktop now...

**Comment by ksubrama** [comment]
its_happening.gif!

**Comment by lamont-granquist** [comment]
gonna at least wait to see if appveyor goes green or not...

**Comment by lamont-granquist** [comment]
well except it looks like chocolatey tests actually hit the service to test installs which is down right now...

**Comment by danielsdeleo** [comment]
PARTY TIME

**Comment by lamont-granquist** [comment]
Shipping

**Pull request merged**